### PR TITLE
Apply customizations to the freshly generated release v0.5 pipelines

### PR DESCRIPTION
**Pull request by simonbaird**
See commit messages for details.

Ref: [EC-819](https://issues.redhat.com/browse/EC-819)

### CODE REVIEW

File: .tekton/cli-build.yaml [hunk 1]
```diff
@@ -52,17 +52,17 @@ spec:
       description: Build dependencies to be prefetched by Cachi2
       name: prefetch-input
       type: string
-    - default: "false"
-      description: Java build
-      name: java
-      type: string
```
Comment (simonbaird) [inline .tekton/cli-build.yaml:58]: Looks like this was removed, see https://github.com/konflux-ci/build-definitions/pull/1325

File: .tekton/cli-build.yaml [hunk 2]
```diff
@@ -52,17 +52,17 @@ spec:
...
       description: Image tag expiration time, time values could be something like 1h, 2d, 3w for hours, days, and weeks, respectively.
       name: image-expires-after
     - default: "false"
       description: Build a source image.
       name: build-source-image
       type: string
+    - default: "false"
+      description: Add built image into an OCI image index
+      name: build-image-index
+      type: string
```
Comment (simonbaird) [inline .tekton/cli-build.yaml:65]: FYI, this becomes the "ALWAYS_BUILD_INDEX" param in the task, so it's not well named.

File: .tekton/cli-v05-pull-request.yaml [hunk 3]
```diff
@@ -18,430 +18,31 @@ metadata:
...
+    - name: image-expires-after
+      value: 5d
+    - name: dockerfile
+      value: Dockerfile.dist
+    - name: path-context
+      value: .
+    - name: prefetch-input
+      value: '[{"type": "gomod"}, {"type": "gomod", "path": "tools/kubectl"}, {"type": "rpm"}]'
     - name: build-source-image
-      params:
```
Comment (simonbaird) [inline .tekton/cli-v05-pull-request.yaml:260]: Reviewer beware, this is not a useful side-by-side diff.

File: .tekton/cli-v05-push.yaml [hunk 4]
```diff
@@ -17,428 +17,31 @@ metadata:
...
+    - name: image-expires-after
+      value: ''
+    - name: dockerfile
+      value: Dockerfile.dist
+    - name: path-context
+      value: .
+    - name: prefetch-input
+      value: '[{"type": "gomod"}, {"type": "gomod", "path": "tools/kubectl"}, {"type": "rpm"}]'
     - name: build-source-image
-      params:
```
Comment (simonbaird) [inline .tekton/cli-v05-push.yaml:257]: Reviewer beware, this is not a useful side-by-side diff.

**Review by lcarva — APPROVED** [review]

**Comment by simonbaird** [comment]
See also #1946 which is related.

**Comment by simonbaird** [comment]
Let's merge, if anything's not quite right we can fix it later.

**Pull request merged**